bin/proving-service/src/proxy/status.rs

```diff
@@ -24,3 +37,96 @@ pub async fn create_status_client(
 
     Ok(StatusApiClient::new(channel))
 }
+
+/// Status of a worker
+#[derive(Debug, Serialize, Deserialize)]
+pub struct WorkerStatus {
+    address: String,
+    is_available: bool,
+    is_healthy: bool,
+    retries: usize,
+}
+
+/// Status of the proxy
+#[derive(Debug, Serialize, Deserialize)]
+pub struct ProxyStatus {
+    version: String,
+    supported_proof_types: ProverTypeSupport,
+    total_workers: usize,
+    busy_workers: usize,
+    workers: Vec<WorkerStatus>,
+}
+
+/// Service that handles status requests
+pub struct StatusService {
```

Comment:
nit: If this is for internal users, could this be part of the control service?

Reply:
Yes, if that's the case then we can merge both services.

Reply:
I would keep them separate as control service would be only for internal users, while status service may be exposed publicly (or at least may be much more open than the control service).